Source/Dafny/AST/DafnyAst.cs

```diff
@@ -2788,6 +2813,9 @@ public class MapType : CollectionType {
         return Domain.MayInvolveReferences || Range.MayInvolveReferences;
       }
     }
+    public override bool DoesNotContainGhostConstraints() {
+      return Range.DoesNotContainGhostConstraints() && Domain.DoesNotContainGhostConstraints();
```

Comment:
Is this covered in the tests?

Reply:
I haven't found a case when the ghost inference did consider a map with a "ghost" subset type either in the Range or its domain as "compilable", so I don't think I can come up with a case where returning true or false here would make a difference in files
Would you like however that I add small unit tests for all of these public functions in `DafnyPipeline.Test`?